packages/mcp-auth/src/auth/mcp-auth-handler.ts

```diff
@@ -0,0 +1,26 @@
+import { type Router } from 'express';
+
+import type { TokenVerifier } from './token-verifier.js';
+
+/**
+ * Defines the contract for a handler that manages the logic for a specific MCPAuth configuration.
+ * This allows for clean separation of logic between legacy and modern configurations.
+ */
+export type MCPAuthHandler = {
```

Comment:
use abstract class to have the full OOP benefit. e.g. can use `instanceof MCPAuthHandler`